### ISSUE TITLE
Actions in e-mail templates have nothing hooked on them, unless WC_Emails constructor is called.

### ISSUE THREAD

**Issue by zachop** [WordImpress/woocommerce-preview-emails#15]
Hi,

You should probably call `WC_Emails:instance();` at some point in `woo-preview-emails.php`.

The reason is that the functions that display the various order sections (e.g. customer details) are hooked to the appropriate actions inside the class constructor (see `woocommerce/includes/class-wc-emails.php`).

As a result, if the class isn't instantiated, you load the email's header, template and footer just fine, but any actions that fire inside the e-mail template have nothing hooked on them.

**Comment by mathetos** [comment]
Hi @zachop thanks for your input. This is a pretty old script that definitely needs to be revisted. I'll take a deeper look and see if I can update it with your suggestions. Thanks!

**Comment by DMDc0de** [comment]
Yes, also after calling `WC_Emails:instance();` , remember to comment out 

```
//wc_get_template( 'emails/email-header.php', array( 'order' => $order, 'email_heading' => $email_heading ) );

do_action( 'woocommerce_email_before_order_table', $order, $sent_to_admin, $plain_text  );

wc_get_template( 'emails/' . $_GET['file'], array( 'order' => $order ) );

//wc_get_template( 'emails/email-footer.php', array( 'order' => $order ) );
```

or header and footer will get duplicated.

**Comment by hypnoboutique** [comment]
Thanks, @zachop and @DMDc0de, this thread helped me a lot.

I also moved everything required from the replacement header and footer files (the css and js) into one file, moved this to a plugin directory (as a plugin!) and made use of the existing header and footer from the wc emails dir.

Oh, also, adding `$("#ordernum").attr("value", getUrlVars()["order"])` somewhere in the ready function repopulates the input box on each load and swapping out `getQueryVariable("order")` for getUrlVars()["order"] will fix the javascript error (missing function)

And I added 'email-order-details.php' and 'email-customer-details.php' to the ignore list (they're template fragments probably new sine this was authored?).

Apologies for the butchery and thanks so much to @mathetos for sharing this code - very helpful and I wish the WC team would incorporate into the core! Wouldn't take long to tidy into a legit plugin but I don't have the time :(
